lib/classes/ConfigSchemaHandler/index.js

```diff
@@ -232,7 +232,29 @@ class ConfigSchemaHandler {
       );
     }
 
-    addPropertiesToSchema(existingEventDefinition.properties[name], configSchema);
+    let definitionToUpdate;
+    if (
+      Object.prototype.hasOwnProperty.call(existingEventDefinition.properties[name], 'properties')
```

Comment:
Let's configure it as `if (existingEventDefinition.properties[name].type === 'object')`. I think it'll be more to the point